subprojects/scala/src/main/java/org/gradle/language/scala/tasks/BaseScalaCompileOptions.java

```diff
@@ -225,4 +227,19 @@ public IncrementalCompileOptions getIncrementalOptions() {
     public void setIncrementalOptions(IncrementalCompileOptions incrementalOptions) {
         this.incrementalOptions = incrementalOptions;
     }
+
+    /**
+     * Keeps Scala compiler daemon alive across builds for faster build times
+     * Legal values:
+     * - false (compiler is kept alive for a session)
+     * - true (compiler is kept alive across builds )
+     */
+     @Input
+    public boolean isKeepAliveAcrossBuilds() {
```

Comment:
❌ The return type should be `Property<KeepAliveOptions>`

where KeepAliveOptions is a new enum with values `SESSION`, and `ACCROSS_BUILDS`.

(also, please use Property.convention() to set up a default value).

Reply:
Thanks @donat . I have re-implemented this as Property<Boolean> in the latest PR.

I could not figure out how a string ("session") in build.gradle will get converted to KeepAliveOptions.SESSION in the scala compile plugin code.
Can you please help me by suggesting an example ?

Reply:
After going through comments from you and @alpar-t I have re-implemented the logic to use `Property<KeepAliveMode>`.
I've used `KeepAliveMode.valueOf()` to validate the string passed in build.gradle file.